### ISSUE TITLE
Support the VSCode extension on the web

### ISSUE THREAD

**Issue by Yokozuna59** [eclipse-langium/langium#1487]
The current Langium VSCode extension doesn't work on the web (https://vscode.dev):

![IMG_0104](https://github.com/eclipse-langium/langium/assets/87622592/be2654d0-6ba0-4287-b910-210d58444f0c)

I think it would be nice to have the extension runnable on the web.

I think this could be applied to both `langium` and `langium-generator`.

---

I used [`langium-sql`](https://github.com/TypeFox/langium-sql) as a reference to make my extension runnable on both.

**Comment by msujew** [comment]
I've also thought about this, but didn't have the time yet to look into this. The main issue is that the grammar language features imports which are a bit difficult to deal with in a web extension.

Any contribution for this is appreciated though.

**Comment by Yokozuna59** [comment]
@msujew I think another issue would be the conflict in the `generator-langium` package between the web and VSCode extension templates. Both of them have common files but are stored in different directories.

Could you share some guidance pointing me in the correct direction so I may be able to work on this?

**Comment by msujew** [comment]
I don't think that these files should be in conflict. The VSCode web extension only needs 4 things:

1. A `browser` entry point in the `package.json` of the extension
2. A corresponding JavaScript bundle that is generated from a `extension-browser.ts` file.
3. A `main-browser.ts` file that represents the browser based language server.
4. A corresponding entry in the bundler script (`esbuild.mjs`) for the `extension-browser.ts` and `main-browser.ts` file.

There is no overlap between these 4 points and the web template AFAIK.

**Comment by msujew** [comment]
Ah, right. We can probably solve that using some logic in the [generator](https://github.com/eclipse-langium/langium/blob/main/packages/generator-langium/src/index.ts). We use that to merge the package.json in the end for example. Alternatively, this would be a good reason to start generating a npm workspace from the yeoman generator. But that would be a larger refactoring that is out of scope for this change.

**Comment by msujew** [comment]
I've created https://github.com/eclipse-langium/langium/issues/1495 to track the yeoman generator refactoring.
